lib/ansible/modules/network/onyx/onyx_aaa_authentication.py

```diff
@@ -0,0 +1,456 @@
+    reset_for: all
+    reset_type: no-unlock
+- name: reset for user x
+  onyx_aaa_authentication:
+    reset_for: user
+    user_name: x
+    reset_type: no-unlock
+- name: configures account settings configs.
+  onyx_aaa_authentication:
+    admin_no_lockout_enabled: yes
+    unknown_account_hash_username_enabled: false
+    unknown_account_no_track_enabled: true
+- name: configures lockout settings
+  onyx_aaa_authentication:
+    lockout_enabled: yes
+    lock_time: 66
+    unlock_time: 66
+    max_failure_account: 55
+"""
+
+RETURN = """
+commands:
+  description: The list of configuration mode commands to send to the device.
+  returned: always
+  type: list
+  sample:
+    - aaa authentication login default <first_method> <second_method> <third_method> <forth_method>
+    - aaa authentication attempts fail-delay <time>
+    - aaa authentication attempts track enable
+    - no aaa authentication attempts track enable
+    - aaa authentication attempts track downcase
+    - no aaa authentication attempts track downcase
+    - aaa authentication attempts reset user <user_name> <reset_type>
+    - aaa authentication attempts reset user <user_name>
+    - aaa authentication attempts reset all
+    - aaa authentication attempts reset all <reset_type>
+    - aaa authentication attempts class-override admin no-lockout
+    - no aaa authentication attempts class-override admin no-lockout
+    - aaa authentication attempts class-override unknown hash-username
+    - no aaa authentication attempts class-override unknown hash-username
+    - aaa authentication attempts class-override unknown no-track
+    - no aaa authentication attempts class-override unknown no-track
+    - aaa authentication attempts lockout enable
+    - no aaa authentication attempts lockout enable
+    - aaa authentication attempts lockout lock-time <lock_time>
+    - aaa authentication attempts lockout unlock-time <unlock_time>
+    - aaa authentication attempts lockout max-fail <max_failure_account>
+"""
+
+import re
+
+from ansible.module_utils.basic import AnsibleModule
+
+from ansible.module_utils.network.onyx.onyx import show_cmd
+from ansible.module_utils.network.onyx.onyx import BaseOnyxModule
+
+
+class OnyxAaaAuthenticationModule(BaseOnyxModule):
+
+    def init_module(self):
+        """ initialize module
+        """
+        element_spec = dict(
+            auth_login_method1=dict(type='str', choices=['local', 'ldap', 'radius', 'tacacs+']),
+            auth_login_method2=dict(type='str', choices=['local', 'ldap', 'radius', 'tacacs+']),
+            auth_login_method3=dict(type='str', choices=['local', 'ldap', 'radius', 'tacacs+']),
+            auth_login_method4=dict(type='str', choices=['local', 'ldap', 'radius', 'tacacs+']),
+            auth_fail_delay_time=dict(type='int'),
+            auth_track_enabled=dict(type='bool'),
+            track_downcase_enabled=dict(type='bool'),
+            reset_for=dict(type='str', choices=['all', 'user']),
```

Comment:
add relations between the reset parameter.
user_name is mandatory when reset_for is user